### PR TITLE
feat(core/server): reject non-grpc requests

### PR DESCRIPTION
**Pull request by i10416**
If **Content-Type** does not begin with "application/grpc", gRPC servers SHOULD respond with HTTP status of 415 (Unsupported Media Type), according to gRPC specification.

In addition, some of gRPC implementations return 405 if request method is not POST as gRPC speccification expects method to be POST.

Note:
there is an ongoing discussion on how to handle invalid requests https://github.com/Nikkei/vessel-manifests/pull/176732

### CODE REVIEW

File: codegen/testing/src/test/scala/hello/world/TestServiceSuite.scala [hunk 1]
```diff
@@ -72,11 +71,48 @@ class TestServiceSuite extends CatsEffectSuite with ScalaCheckEffectSuite {
     }
   }
 
-  test("Routes returns missing method") {
+  test("Routes returns 405 Method Not Allowed with Allow header on GET requests") {
```
Comment (hamnis) [inline codegen/testing/src/test/scala/hello/world/TestServiceSuite.scala:74]: perhaps we also should test a variant with application/grpc+proto ?

Comment (i10416) [reply codegen/testing/src/test/scala/hello/world/TestServiceSuite.scala:74]: Thank you. I added sanity check for well-known grpc content type.

https://github.com/http4s/http4s-grpc/pull/147/files#diff-c409a3b6f497230f06fc37ad92f044801b41c53b33e22b26d03cda5a932d8a79R78-R84

**Review by hamnis — APPROVED** [review]

File: core/src/main/scala/org/http4s/grpc/ServerGrpc.scala [hunk 2]
```diff
@@ -15,6 +17,16 @@ import java.util.concurrent.TimeoutException
 import scala.concurrent.duration._
 
 object ServerGrpc {
+  def precondition[F[_]: Monad]: HttpRoutes[F] = HttpRoutes.of[F] {
+    case req if req.method =!= Method.POST =>
```
Comment (danicheg) [inline core/src/main/scala/org/http4s/grpc/ServerGrpc.scala:21]: For the sake of saving a few nanoseconds, I think plain `==` will work finely here

**Review by danicheg — APPROVED** [review]
Thanks! LGTM

**Pull request merged**